docs/guides/routing.rst

```diff
@@ -20,47 +20,120 @@ an empty segments array.
-passed in as the first argument.
-
-The following code registers a page handler for "blog" URLs and shows how one might route
-the request to a resource view.
+Elgg offers a facility to manage your plugin pages via custom routes, enabling URLs like ``http://yoursite/my_plugin/section``.
+You can register a new route using ``elgg_register_route()`, or via ``routes`` config in ``elgg-plugin.php``.
+Routes map to resource views, where you can render page contents.
 
 .. code-block:: php
 
-   elgg_register_page_handler('blog', 'blog_page_handler');
+	// in your 'init', 'system' handler
+	elgg_register_route('my_plugin:section' [
+		'path' => 'my_plugin/section/{guid}/{subsection?}',
+		'resource' => 'my_plugin/section',
+		'requirements' => [
+			'guid' => '\d+',
+			'subsection' => '\w+',
+		],
+	]);
 
-   function blog_page_handler(array $segments) {
-        // if the URL is http://example.com/elgg/blog/view/123/my-blog-post
-        // $segments contains: ['view', '123', 'my-blog-post']
+	// in my_plugin/views/default/resources/my_plugin/section.php
+	$guid = elgg_extract('guid', $vars);
+	$subsection = elgg_extract('subsection', $vars);
 
-        $subpage = elgg_extract(0, $segments);
-        if ($subpage === 'view') {
+	// render content
 
-            // use a view for the page logic to allow other plugins to easily change it
-            $resource = elgg_view_resource('blog/view', [
-                'guid' => (int)elgg_extract(1, $segments);
-            ]);
+In the example above, we have registered a new route that is accessible via ``http://yoursite/my_plugin/section/<guid>/<subsection>``.
+Whenever that route is accessed with a required ``guid`` segment and an optional ``subsection`` segment, the router
+will render the specified ``my_plugin/section`` resource view and pass the parameters extracted from the URL to your
+resource view with ``$vars``.
 
-            return elgg_ok_response($resource);
-        }
 
-        // redirect to a different location
-        if ($subpage === '') {
-            return elgg_redirect_response('blog/all');
-        }
+Routes names
+------------
 
-        // send an error page
-        if ($subpage === 'owner' && !elgg_entity_exists($segments[1])) {
-            return elgg_error_response('User not found', 'blog/all', ELGG_HTTP_NOT_FOUND);
-        }
-        
-        // ... handle other subpages
-   }
+Route names can then be used to generate a URL:
+
+.. code::php
+
+	$url = elgg_generate_url('my_plugin:section', [
+		'guid' => $entity->guid,
+		'subsection' => 'assets',
+	]);
+
+
+The route names are unique across all plugins and core, so another plugin can override the route by registering different
+parameters to the same route name.
+
+Route names follow a certain convention and in certain cases will be used to automatically resolve URLs, e.g. to display an entity.
+
+The following conventions are used in core and recommended for plugins:
```

Comment:
Thank you for coming up with this and documenting it.

Reply:
I am thinking ahead for getting rid of bad stuff like `handler` in entity menu, and all the endless `entity:url` hooks we have to write.